src/components/button/index.tsx

```diff
@@ -0,0 +1,49 @@
+import { Component, Event, EventEmitter, h, Listen, Prop } from "@stencil/core"
+import { Color, Expand, Fill } from "../../model"
+
+@Component({
+	tag: "smoothly-button",
+	styleUrl: "style.css",
+	scoped: true,
+})
+export class SmoothlyButton {
+	@Prop({ reflect: true }) color: Color | undefined
+	@Prop({ reflect: true }) expand: Expand
+	@Prop({ reflect: true }) fill: Fill
+	@Prop({ reflect: true }) disabled = false
+	@Prop({ reflect: true }) type: "link" | "button" = "button"
+	@Prop() link?: string
+	@Prop() download?: boolean
+	@Event() onClick: EventEmitter<UIEvent>
+
+	@Listen("click")
+	clicked(e: UIEvent) {
+		this.onClick.emit(e)
+	}
```

Comment:
You don't need to reinvent click events. When an element is clicked a bubble event will automatically bubble up.